api/infrastructure/parallelcluster-api.yaml

```diff
@@ -221,6 +221,93 @@ Resources:
       ImageTestsConfiguration:
         ImageTestsEnabled: false
 
+  EcrImageDeletionLambda:
+    Condition: DoNotUseCustomEcrImageUri
+    Type: AWS::Lambda::Function
+    Properties:
+      Code:
+        ZipFile: |
+          import cfnresponse
+          import boto3
+
+          def get_tag_from_image_arn(image_arn):
+              """ Returns the Docker image tag given an ImageBuilder image ARN
+
+              Arn format: arn:aws:imagebuilder:${AWS::Region}:${AWS::AccountId}:image/.../${Version}/${BuildNumber}
+              Tag format: ${Version}-${BuildNumber}
+              """
+              parts = image_arn.split('/')
+              version = parts[-2]
+              build_number = parts[-1]
+              return '{}-{}'.format(version, build_number)
+
+          def handler(event, context):
+              print(event)
+              print('boto version {}'.format(boto3.__version__))
+
+              response_data = {}
+              response_status = cfnresponse.SUCCESS
+
+              if event['RequestType'] == 'Create':
+                  response_data['Message'] = 'Resource creation successful!'
+              elif event['RequestType'] == 'Update':
+                  response_data['Message'] = 'Resource update successful!'
+              elif event['RequestType'] == 'Delete':
+                  ecr = boto3.client('ecr')
+
+                  ecr_repository_name = event['ResourceProperties']['EcrRepositoryName']
+                  tag = get_tag_from_image_arn(event['ResourceProperties']['ImageToDeleteArn'])
+
+                  ecr.batch_delete_image(repositoryName=ecr_repository_name, imageIds=[{'imageTag': tag}])
+
+                  response_data['Message'] = 'Resource deletion successful!'
+
+              cfnresponse.send(event, context, response_status, response_data)
+
+      Handler: index.handler
+      Runtime: python3.7
+      Role: !GetAtt EcrImageDeletionLambdaRole.Arn
+
+  EcrImageDeletionLambdaRole:
+    Condition: DoNotUseCustomEcrImageUri
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Policies:
+        - PolicyName: LoggingPolicy
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogGroup
```

Comment:
granting CreateLogGroup permissions will cause this lambda to create a log group that will not be deleted. Shall we explicitly create a log group in the template?